### PR TITLE
CMake updates

### PR DESCRIPTION
**Pull request by mahrud**
- print list of optional libraries in cmake
- minor addition in INSTALL-CMake.md
- added msolve as a library for cmake

### CODE REVIEW

**Comment by mahrud** [comment]
Because I disabled them with `[skip ci]`.

**Comment by DanGrayson** [comment]
I don't see `[skip ci]` anywhere.

Don't we have to test whether your new code works?

**Comment by mahrud** [comment]
It's in the commit message in https://github.com/Macaulay2/M2/pull/2842/commits/b528ef381523276f2e4b656afa990f3df24e3472:
```
added msolve as a library for cmake
[skip ci]
```

Have you looked at the changes? There's nothing that needs testing because none of it would actually be run on github actions.

**Comment by mahrud** [comment]
@mikestillman whenever you need to test things with msolve, this should have everything you need. You can run `ninja build-msolve`, which I'll add to INSTALL-CMake.md in a second.

**Comment by moorewf** [comment]
@mahrud the msolve build seems to be having trouble finding gmp using cmake.  Any suggestions?

**Comment by mahrud** [comment]
What do you mean by finding gmp using cmake? The msolve configure script isn't designed to accept any of `--with-gmp, --with-mpfr, --with-flint`, so if msolve can't find gmp there's little that cmake can do about it.

You could try a fix like this:
```diff
diff --git a/M2/cmake/build-libraries.cmake b/M2/cmake/build-libraries.cmake
index b68fdd0e9c..a8170c17fa 100644
--- a/M2/cmake/build-libraries.cmake
+++ b/M2/cmake/build-libraries.cmake
@@ -552,9 +552,9 @@ ExternalProject_Add(build-msolve
             COMMAND ${CONFIGURE} --prefix=${M2_HOST_PREFIX}
                       #-C --cache-file=${CONFIGURE_CACHE}
                       ${shared_setting}
-                      CPPFLAGS=${CPPFLAGS}
+                      "CPPFLAGS=${CPPFLAGS} -I${MP_INCLUDE_DIRS}"
                       CFLAGS=${CFLAGS}
-                      LDFLAGS=${LDFLAGS}
+                      "LDFLAGS=${LDFLAGS} -L${MP_LIBRARY_DIRS}"
                       CC=${CMAKE_C_COMPILER}
                      "OPENMP_CFLAGS=${OpenMP_C_FLAGS} ${OpenMP_C_LDLIBS}"
   BUILD_COMMAND     ${MAKE} -j${PARALLEL_JOBS}
```
But if you run into the same issue with mpfr and flint you should file an issue in the msolve repository about this.

**Comment by moorewf** [comment]
Thanks, I will try something like this.  I tried to build using cmake (build-msolve) on my machine and their configure script seemed to be having trouble finding gmp.

**Comment by DanGrayson** [comment]
Mahrud, can you tell what is wrong with this build?

https://github.com/Macaulay2/M2/actions/runs/5016638199/jobs/8993737981

**Comment by mahrud** [comment]
> Mahrud, can you tell what is wrong with this build?
> 
> https://github.com/Macaulay2/M2/actions/runs/5016638199/jobs/8993737981

Yes, I forgot to comment out a [line](https://github.com/Macaulay2/M2/pull/2846/commits/988d83f14326fec833f25018fbd6cb05f98dc0ac). Should be fixed by #2846.
